spec/controllers/invoices/pdf_controller_spec.rb

```diff
@@ -9,44 +9,34 @@
     sign_in employee
   end
 
-  describe 'GET #new' do
+  describe 'GET #show' do
     let(:pdf_double) { instance_double Invoices::PDF }
+    let(:pdf_data) { double('data') }
 
     before do
       allow(Invoices::PDF).to receive(:new).and_return pdf_double
+      allow(pdf_double).to receive(:read).and_return pdf_data
       allow(pdf_double).to receive(:persisted?).and_return true
       allow(pdf_double).to receive(:persisted_pdf_path).and_return 'persisted.pdf'
     end
 
     it 'returns pdf path' do
-      expect(controller).to receive(:send_file).with('persisted.pdf', filename: 'Rechnung Hans 2017-09-04.pdf')
+      expect(controller).to receive(:send_data).with(pdf_data, filename: 'Rechnung Hans 2017-09-04.pdf')
```

Comment:
RSpec/MessageSpies: Prefer `have_received` for setting message expectations. Setup `controller` as a spy using `allow` or `instance_spy`.